### PR TITLE
Added reparameterization and new loss function

### PR DESCRIPTION
**Pull request by httpjesse**

### CODE REVIEW

File: scinet/scinet.py [hunk 1]
```diff
@@ -47,73 +50,118 @@ def __init__(self, hyp):
...
+            # for i in tqdm(range(0, testSize, batchSize)):
+            for i in range(0, testSize, batchSize):
 
                 observationBatch = observations[i:i + batchSize].to(self.device)
                 answersBatch = answers[i:i + batchSize].to(self.device)
                 questionBatch = questions[i:i + batchSize].to(self.device)
 
-                outputs = self(observationBatch, questionBatch)
-                loss = self.lossFunct(outputs, answersBatch)
+                mu, sig, Z, outputs = self(observationBatch, questionBatch, verbose=verbose)
```
Comment (nmdickso) [inline scinet/scinet.py:157]: How are you guys normally inspecting the latent neuron activations? I'm realizing now that in neither of these versions is the activation (Z) actually returned, in any way.

In `copernicus` (in the test method) I'm saving the latent activations as we train, then returning them alongside the loss, to analyze. Should we be doing something like that here as well? I'm just curious what you all are doing right now, without that.

**Review by nmdickso — APPROVED** [review]
This all looks good enough to merge, I recommend everyone test it out as soon as possible.

I'd also like a response on my comment below, but it can be addressed in a later pr, we don't need to wait for it to merge this.

File: scinet/scinet.py (hunk 1 above)
Comment (nmdickso) [reply scinet/scinet.py:157]: @keyblaede

Comment (httpjesse) [reply scinet/scinet.py:157]: I have not been examining activation as my results haven't been strong enough to bother yet. I am not sure how to go about doing this, but I an down to do it any way you would like and I can implement it if you would like.

Comment (httpjesse) [reply scinet/scinet.py:157]: @nmdickso

Comment (nmdickso) [reply scinet/scinet.py:157]: Okay, don't worry about it, I'll work with Veronica to implement something, probably in a similar fashion to how I've done it for `copernicus`, it's mostly just for visualizations anyways.
Thanks
